controller/api/destination/external-workload/endpoints_controller.go

```diff
@@ -0,0 +1,513 @@
+		AddFunc:    ec.updateService,
+		DeleteFunc: ec.updateService,
+		UpdateFunc: func(_, newObj interface{}) {
+			ec.updateService(newObj)
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = k8sAPI.ExtWorkload().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			ew, ok := obj.(*ewv1alpha1.ExternalWorkload)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: expected *v1alpha1.ExternalWorkload, got %#v", obj)
+				return
+			}
+
+			if len(ew.Spec.WorkloadIPs) == 0 {
+				ec.log.Debugf("skipping ExternalWorkload event: %s/%s has no IP addresses", ew.Namespace, ew.Name)
+				return
+			}
+
+			if len(ew.Spec.Ports) == 0 {
+				ec.log.Debugf("skipping ExternalWorkload event: %s/%s has no ports", ew.Namespace, ew.Name)
+				return
+			}
+
+			services, err := ec.getSvcMembership(ew)
+			if err != nil {
+				ec.log.Errorf("failed to get service membership for %s/%s: %v", ew.Namespace, ew.Name, err)
+				return
+			}
+
+			for svc := range services {
+				ec.enqueueUpdate(svc)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			var ew *ewv1alpha1.ExternalWorkload
+			if ew, ok := obj.(*ewv1alpha1.ExternalWorkload); ok {
+				services, err := ec.getSvcMembership(ew)
+				if err != nil {
+					ec.log.Errorf("failed to get service membership for %s/%s: %v", ew.Namespace, ew.Name, err)
+					return
+				}
+
+				for svc := range services {
+					ec.enqueueUpdate(svc)
+				}
+				return
+			}
+
+			tomb, ok := obj.(cache.DeletedFinalStateUnknown)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: couldn't retrieve obj from DeletedFinalStateUnknown %#v", obj)
+				return
+			}
+
+			ew, ok = tomb.Obj.(*ewv1alpha1.ExternalWorkload)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: DeletedFinalStateUnknown contained object that is not a v1alpha1.ExternalWorkload %#v", obj)
+				return
+			}
+
+			services, err := ec.getSvcMembership(ew)
+			if err != nil {
+				ec.log.Errorf("failed to get service membership for %s/%s: %v", ew.Namespace, ew.Name, err)
+				return
+			}
+
+			for svc := range services {
+				ec.enqueueUpdate(svc)
+			}
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			old, ok := oldObj.(*ewv1alpha1.ExternalWorkload)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: expected *v1alpha1.ExternalWorkload, got %#v", oldObj)
+			}
+
+			updated, ok := newObj.(*ewv1alpha1.ExternalWorkload)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: expected *v1alpha1.ExternalWorkload, got %#v", newObj)
+			}
+
+			// Ignore resync updates. If nothing has changed in the object, then
+			// the update processing is redudant.
+			if old.ResourceVersion == updated.ResourceVersion {
+				ec.log.Tracef("skipping ExternalWorkload resync update event")
+				return
+			}
+
+			// Compute whether any labels have changed in between updates. If
```

Comment:
There's a lot of logic here to try to determine if the change to the ExternalWorkload is a material difference (and therefore if endpoints should be recomputed for any services).  It would simplify things if we enqueued the service updates for any change here.  Do you think this would cause problematic load? It would only trigger updates for services which select the old or new ExternalWorkload.